src/flash.sol

```diff
@@ -158,7 +161,7 @@ contract DssFlash is IERC3156FlashLender, IVatDaiFlashLender {
         dai.transferFrom(address(receiver), address(this), total);
         daiJoin.join(address(this), total);
         vat.heal(rad);
-        vat.move(address(this), vow, mul(fee, RAY));
+        require(vat.dai(address(this)) >= add(prev, fee), "DssFlash/ceiling-exceeded");
```

Comment:
I don't think the error message here is quite correct...

Reply:
I'm also convinced this check is necessary. It wouldn't hurt if e.g. someone calls `accrue` during a flash loan.

Reply:
Yes the message is wrong and the check is necessary in this one but not in the other function as daiJoin.join will fail if can't be done for the total.

Reply:
To clarify the require is necessary on the vat flash loan as the only constraint is the `heal` which is only for the amount lent. However on the erc20 one the transferFrom will require the whole amount.